packages/config/src/bin/dev.webpack.config.ts

```diff
@@ -1,18 +1,22 @@
-const config = require('@redhat-cloud-services/frontend-components-config');
-const commonPlugins = require('./webpack.plugins.js');
-const fecConfig = require(process.env.FEC_CONFIG_PATH);
+import { ProxyConfigArrayItem } from 'webpack-dev-server';
+import config, { FrontendEnv } from '../lib';
+import FECConfiguration from '../lib/fec.config';
+import commonPlugins from './webpack.plugins';
+const fecConfig: FECConfiguration = require(process.env.FEC_CONFIG_PATH!);
+
+type Configuration = import('webpack').Configuration;
 
 const isBeta = process.env.BETA === 'true';
 
-function parseRegexpURL(url) {
-  return isBeta ? [new RegExp(`/beta${url.toString()}`), new RegExp(`/preview${url.toString()}`)] : new RegExp(url.toString());
+function parseRegexpURL(url: RegExp) {
+  return isBeta ? [new RegExp(`/beta${url.toString()}`), new RegExp(`/preview${url.toString()}`)] : [new RegExp(url.toString())];
 }
 
-function createAppUrl(appUrl) {
+function createAppUrl(appUrl: string | string[] | (string | RegExp)[]) {
   if (Array.isArray(appUrl)) {
     return appUrl
       .map((url) => {
-        if (typeof url === 'object') {
+        if (url instanceof RegExp) {
```

Comment:
:+1: for the tighter check here. Hopefully no one is relying on passing in a native object.

Reply:
I don't this so. I would not work.